plugins/catalog-backend/src/util/opentelemetry.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Span, SpanOptions, SpanStatusCode, Tracer } from '@opentelemetry/api';
+import { parseEntityRef } from '@backstage/catalog-model';
+
+export const TRACER_ID = 'backstage-plugin-catalog-backend';
+
+export function addEntityAttributes(span: Span, entityRef: string) {
+  try {
+    const fields = parseEntityRef(entityRef);
```

Comment:
> The triplet you get out of this will not be the same as the actual values inside the entity body

If they aren't processing will fail in `runValidateStep` right? It checks `if (stringifyEntityRef(entity) !== context.entityRef) {`, so if the values in the body don't match the ref it's all thrown out

> Do you have specific need to have the three as separate attributes?

I want to be able to have stats on processing time for different kinds of entity, at the least. I think in general it makes sense to use the raw values, rather than stringified combinations. We can always combine stuff for display etc, but having the data in individual keys is always more useful.

> It's a bit harsh to fail the entire span just because something went wrong with this parsing, too
Fair. I can look at changing that..
Or to be fair, I think all the places we're getting this we also have the actual item. I could change this to pull directly from the entity itself (if the attributes exist etc, make sure it can't fail)?

Reply:
Ohhh. I see what you mean. I'll have a think about this

Reply:
I've swapped this over to use the entity itself :)